API/c/Inc/interrupt_in.h

```diff
@@ -0,0 +1,52 @@
+#ifndef __INTERRUPT_IN_H
+#define __INTERRUPT_IN_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "pin_defs.h"
+
+typedef enum {
+    INTERRUPT_RISING = GPIO_MODE_IT_RISING,
+    INTERRUPT_FALLING = GPIO_MODE_IT_FALLING,
+    INTERRUPT_RISING_FALLING = GPIO_MODE_IT_RISING_FALLING
+} interrupt_mode;
+
+ /** Configures GPIO pin for interruptin
+ *
+ * Defaults to no pull type and both edge detection
+ *
+ * @param pin Pin def external to board
+ * @param function Function linked to interrupt
+ */
+void interruptin_init(pin_name pin, void (*function)()); // void (*function)(void)
```

Comment:
There should be a C++ wrapper for this eventually

Reply:
Isn't there a C++ interrupts?

Reply:
You right. I forgot I did the C version to not have to worry about any wonky c++ shit happening since this was very last minute at comp